hash.py

```diff
@@ -0,0 +1,23 @@
+class Hash(object):
+    def __init__(self, buckets):
+        self.heap = []
+        for i in range(0, buckets):
+            self.heap.append([])
+
+    def set(self, key, val):
+        """store the given val using the given key"""
+        tup = (key, val)
+        self.heap[self.hash(key)].append(tup)
```

Comment:
What happens if the key you have passed in was already in the hash table?  What happens if it was used to store a *different* value than what is passed in this time?